app/code/Magento/Widget/Test/Mftf/ActionGroup/AdminCreateAndDeleteMultipleLayoutWidgetActionGroup.xml

```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+<actionGroups
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="AdminCreateAndDeleteMultipleLayoutWidgetActionGroup">
```

Comment:
Hi @saphaljha. Great work! Thank you for the coverage. 
We may improve this action group by splitting it into more atomic action groups. The main reason of it is the following. The purpose of the current action group is very specific and it might not be reused in many cases. In the meantime, we could split this huge action group into pieces like action group that creates widget, action group that deletes widget, action group that asserts number elements. In this case the smaller action groups will be much more reusable. Also, it's a good idea to double-check if we could reuse the [existing](https://github.com/magento/magento2/blob/16e6fdace4f1e297a43fe396b4d8f45432e82702/app/code/Magento/Widget/Test/Mftf/ActionGroup/AdminCreateWidgetActionGroup.xml#L11) action groups for adding/removing widgets. 

Hope it makes sense. 

Thank you!

Reply:
Thank you @rogyar ,
I have updated it MFTF as per your suggestions, Please check and let me know if there is anything.